ExternalTest/rr_sim_tests/rr_sim_tests/utils/base_subscriber.py

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env python3
+# Copyright 2020-2023 Rapyuta Robotics Co., Ltd.
+
+import asyncio
+import time
+
+# rclpy
+import rclpy
+
+# other ros
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.node import Node
+from rclpy.qos import QoSDurabilityPolicy, QoSReliabilityPolicy, QoSHistoryPolicy, QoSProfile
+
+# rr_sim_tests
+from rr_sim_tests.utils.wait_for_topics import WaitForTopics
+
+# Data polling rate
+SUBSCRIBER_DEFAULT_RATE = 10.0 #Hz
+SUBSCRIBER_DEFAULT_QOS = QoSProfile(
+    reliability=QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_RELIABLE,
+    history=QoSHistoryPolicy.RMW_QOS_POLICY_HISTORY_KEEP_ALL,
+    depth=10
+)
+
+class BaseSubscriber(Node):
+    def __init__(self, in_robot_namespace: str, 
+                       in_topic_name: str,
+                       in_subscriber_rate=SUBSCRIBER_DEFAULT_RATE):
+        super().__init__(
+            node_name=f"{in_topic_name.replace('/','_')}_subscriber", namespace=in_robot_namespace
+        )
+
+        # https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/node_impl.hpp
+        # https://github.com/ros2/rclcpp/issues/1767
+        self._full_topic_name = None
```

Comment:
if this is a common feature for both pub and sub, should be separated

Reply:
You meant grouped?, I so added `base_pub_sub_node.py`